### ISSUE TITLE
libgd 2.2.4: Compilation failure on illumos in gdkanji.c due to constness of iconv argument.

### ISSUE THREAD

**Issue by alarcher** [libgd/libgd#392]
On illumos iconv.h defines two versions of iconv(): a non-const version if the XPG6 standard is enforced, and a const version otherwise.

Reference: http://src.illumos.org/source/xref/illumos-gate/usr/src/head/iconv.h

The attached patch enforces the constness consistently.
Thank you.

[007-iconv.patch.txt](https://github.com/libgd/libgd/files/1065900/007-iconv.patch.txt)

**Comment by vapier** [comment]
i don't think that patch is the right approach.  we don't want to sprinkle defines like sun throughout the source, and the const issue shows up in other implementations as well.  and i think this breaks non-sun systems because `ICONV_CONST` isn't defined.

we should make this into a configure test i think.  longer term we could kill this file entirely (via issue #393).

**Comment by alarcher** [comment]
OK, you can close the issue then.
As long as you decide which standard you want to comply with there is no problem.

**Comment by vapier** [comment]
does adding this to the top of the file fix your build issues ?

`#define _XOPEN_SOURCE 700`

**Comment by alarcher** [comment]
Yes anything >=600 (to satisfy XPG6) and together with passing -DHAVE_ICONV_T_DEF=1 makes the build succeed.
The definition HAVE_ICONV_T_DEF is the one difference I noticed between CMake and Autotools builds on my system, the rest is similar otherwise.
I removed the patch and passed -DXOPEN_SOURCE=600 -DHAVE_ICONV_T_DEF=1 instead.
With Autotools iconv.m4 is used for setting HAVE_ICONV_T_DEF, no problem there.
Thank you.
